### PR TITLE
Update to use ArcGIS runtime update 10

### PR DESCRIPTION
**Pull request by puneet-pdx**
- Android Gradle Plugin Version - 4.1.2
- Min SDK - 23 required by ArcGIS runtime
- ArcGIS runtime version - 100.10.0
- googleLocationServicesVersion - 17.1.0
- materialLibraryVersion
- appcompatLibraryVersion.       

- remove explicit addition of multiDex support (not needed with min sdk version 23)
>If your minSdkVersion is set to 21 or higher, multidex is enabled by default and you do not need the multidex library
https://developer.android.com/studio/build/multidex.html#mdex-gradle
- remove useProguard (not needed with android tools version > 3.5)
https://developer.android.com/studio/build/shrink-code.html#enable

### CODE REVIEW

**Review by sorenoid — APPROVED** [review]
👍

**Review by esreli — CHANGES_REQUESTED** [review]
The changes you've made to the project look good to me, thank you for updating the project and its dependencies accordingly. It appears there doesn't need to be any code changes, nice!

As for the doc, with the launch of the new site there appears to be a number of broken links.

### README

- [License your app](https://github.com/Esri/nearby-android/tree/puneet/updateToUseU10#licensing) leads to `404`. Why don't we point to the android SDK's [License and deployment](https://developers.arcgis.com/android/license-and-deployment/) page?

### docs/README

- [mapless app pattern](https://github.com/Esri/nearby-android/tree/puneet/updateToUseU10/docs#device-location) leads to `404`. I'm not really sure what to link to here, I'm not sure if a mapless pattern is discussed in the new doc. You can probably simply remove this hyperlink, unless you have a good link?

**Comment by puneet-pdx** [comment]
@esreli I have addressed your comments. Please have another look.

File: README.md [hunk 1]
```diff
@@ -110,7 +110,7 @@ Unless required by applicable law or agreed to in writing, software distributed
 
 A copy of the license is available in the repository's [LICENSE](LICENSE) file.
 
-For information about licensing your deployed app, see [License your app](https://developers.arcgis.com/android/guide/license-your-app.htm).
+For information about licensing your deployed app, see [License your app](https://developers.arcgis.com/android/license-and-deployment).
```
Comment (esreli) [inline README.md:113]:
```suggestion
For information about licensing your deployed app, see [License your app](https://developers.arcgis.com/android/license-and-deployment/).
```

**Review by esreli — APPROVED** [review]
LGTM 👍

**Pull request merged**